### ISSUE TITLE
No Tray Icon on 18.04

### ISSUE THREAD

**Issue by pwd-github** [muammar/mkchromecast#163]
Anyone else experiencing this? Anyone know a fix? Even with Top Icons Plus it's missing.

**Comment by ygmarchi** [comment]
Yes, I see the same problem

**Comment by pwd-github** [comment]
Mines come back this morning.

**Comment by muammar** [comment]
I installed 18.04 in a Vbox and I see the problem, too. Do you still see icons for applications like Dropbox?

**Comment by kocoai** [comment]
No tray for me too: v0.3.8.1

**Comment by leamas** [comment]
Basically, I don't think using the tray icon for new code is a good idea. At least on the LInux side, both Fedora/Gnome3 and Ubuntu been moving away from the tray icon/statusicon abstraction since long. Gnome3 has now dropped the legacy support for it, and Ubuntu will be using Gnome3 from now on. So, a perhaps better idea is to just a regular GUI app with the necessary controls.

And, on linux, add systemctl --user support  to manage the service. Bt that's another story...

**Comment by muammar** [comment]
@leamas you are right. I have just read that Gnome dropped the system tray. Creating a GUI makes lots of sense now. I have started [something](https://github.com/muammar/mkchromecast/tree/ui) but my knowledge on UIs is limited so it will take some time for me. I have just logged in and I saw that Dropbox has a working icon but I suppose that is something rather complicated to make it work in Gnome.

Now, this is more of a personal opinion: 

<rant>Linux usually sucks when it comes to the desktop UX. IGnome used to be cool, and after 2011 I ditched it in favor of something more useful: a windows manager (awesome WM). I don't see the point why they dropped the system tray. I use Linux for my daily workflow, am Debian Developer, and pro Linux. But, Windows and macOS, regardless whether they suck as OS or not, still use a system tray. In addition, it is very well known, whether or not one agrees, that macOS UX is "good" or functional. And, they have a system tray. So why Gnome faced it out? No f.... idea... I am sure there will be an extension for having a system tray... that is so sad and one of the reasons why Linux has not become mainstream. </rant>

**Comment by leamas** [comment]
https://www.omgubuntu.co.uk/2017/09/will-you-miss-gnome-legacy-tray

My gut reaction is the same as yours. That said, I've stick to using gnome3 and found that the designers has developed a really nice desktop over time. So, I somehow respect their judgment and I'm prepared to listen to them.

What about developing a regular gui app instead? A main window with all the bells and whistles mkchromecast needs? It could be minimized, in in that shape work more or less as a status icon. That's sort of the common denominator, I guess - it should work everywhere.

**Comment by muammar** [comment]
> What about developing a regular gui app instead? A main window with all the bells and whistles mkchromecast needs? It could be minimized, and in that shape work more or less as a status icon. That's sort of the common denominator, I guess - it should work everywhere.

I think the same. A window that does all these things, that's the way to go. As I mentioned, I started a GUI, but I am afraid that it will look very ugly at the beginning as I am too new to building these interfaces.

**Comment by baselbers** [comment]
Can you give us an update about this issue? Can we expect a solution anytime soon?

**Comment by yerbestpal** [comment]
This is also relevant for me. No tray icon But notifications are still present.

**Comment by localredhead** [comment]
I was having this problem too... but it's because I have a dark theme and the icon defaults to black.

![screenshot from 2018-10-17 11-52-40](https://user-images.githubusercontent.com/152448/47099525-79808200-d203-11e8-8749-4ea24c08e5f0.png)

I found this out by randomly clicking on my status bar area and saw a drop down appear.

I fixed this by clicking on **Preferences** and changing the Icon color.  I then had to quite mkchromecast and restart, and now I have the system tray icon:  
![screenshot from 2018-10-17 11-57-37](https://user-images.githubusercontent.com/152448/47099723-e5fb8100-d203-11e8-9341-e1fb924e5bb6.png)

Hope this helps

**Comment by dsego** [comment]
Thanks @localredhead, this was my issue as well, but now the icon is tiny. This is on Ubuntu 18.10 with Gnome 200% scaling (2560x1440 screen). But at least I can see it's there.

![screenshot from 2018-12-01 14-36-54](https://user-images.githubusercontent.com/578557/49328805-99cb8c80-f576-11e8-881f-948e0bbb3c98.png)

**Comment by ericpulvino** [comment]
Something that integrates as a shell extension might make a ton of sense. I use this shell extension all the time to switch audio sinks https://extensions.gnome.org/extension/906/sound-output-device-chooser/
Not sure if there would be a way to integrate like this and punt the settings window development for later :)

**Comment by pwd-github** [comment]
I posted a possible solution to this as #223

**Comment by muammar** [comment]
@ericpulvino that sounds cool, but I have no idea how to do that. Have you ever created an extension for Gnome?

**Comment by fabrixx** [comment]
i not have that icon, i clicked all over my top bar with Gnome 3.32/Arch

**Comment by pwd-github** [comment]
> i not have that icon, i clicked all over my top bar with Gnome 3.32/Arch

Do you have KStatusNotifierItem/Appindicator Support or Top Icons Plus installed?

**Comment by fabrixx** [comment]
With TopIcons Redux & device choser addon i see icon and is more confortable that from terminal.
i'm working to setup preferences to reduce delay

**Comment by Bewbman** [comment]
/usr/share/mkchromecast/mkchromecast/systray.py

Delete or comment out line 169:
self.ag = QtWidgets.QActionGroup(self, exclusive=True)

Maybe someone more aware in Python can tell me what this does, IDK enough about but I do know that after Python 5.14 'exclusive' is no longer part of the language. Regardless I'm still trying to figure out how to get it to pair now lol, when I search for streaming devices it doesn't find anything. Will keep updated.
